pkg/remote/utils.go

```diff
@@ -0,0 +1,286 @@
+	_, err = node.RunEveCommand(command)
+	return err
+}
+
+// WaitForAppStart waits for an app to start on the EVE node
+func (node *RemoteNode) WaitForAppStart(appName string, timeoutSeconds uint) error {
+	start := time.Now()
+	for {
+		state, err := node.GetAppState(appName)
+		if err != nil {
+			return err
+		}
+
+		if strings.ToLower(state) == "running" {
+			return nil
+		}
+
+		if time.Since(start) > time.Duration(timeoutSeconds)*time.Second {
+			return fmt.Errorf("timeout waiting for app %s to start", appName)
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// WaitForSSH waits for the SSH connection to be established to the app VM that
+// is running on the EVE node
+func (node *RemoteNode) WaitForSSH(timeoutSeconds uint) error {
+	start := time.Now()
+	for {
+		_, err := node.AppSSHExec("echo")
+		if err == nil {
+			return nil
+		}
+
+		if time.Since(start) > time.Duration(timeoutSeconds)*time.Second {
+			return fmt.Errorf("timeout waiting for SSH connection")
+		}
+
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// StopAndRemoveApp stops and removes an app from the EVE node
+func (node *RemoteNode) StopAndRemoveApp(appName string) error {
+	if err := node.Controller.PodStop(appName); err != nil {
+		return err
+	}
+
+	if _, err := node.Controller.PodDelete(appName, true); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetNodeIP gets the IP address of the app running the EVE node
+func (node *RemoteNode) GetNodeIP() error {
+	if node.Edgenode.GetRemoteAddr() == "" {
+		eveIPCIDR, err := node.Tc.GetState(node.Edgenode).LookUp("Dinfo.Network[0].IPAddrs[0]")
+		if err != nil {
+			return err
+		}
+
+		ip := net.ParseIP(eveIPCIDR.String())
+		if ip == nil || ip.To4() == nil {
+			return fmt.Errorf("failed to parse IP address: %s", eveIPCIDR.String())
+		}
+
+		node.IP = ip.To4().String()
+		return nil
+	}
+
+	node.IP = node.Edgenode.GetRemoteAddr()
+	return nil
+}
+
+// GetAppState gets the state of an app running on the EVE node
+func (node *RemoteNode) GetAppState(appName string) (string, error) {
+	ctrl, err := controller.CloudPrepare()
+	if err != nil {
+		return "", fmt.Errorf("fail in CloudPrepare: %w", err)
+	}
+
+	state := eve.Init(ctrl, node.Edgenode)
+	if err := ctrl.InfoLastCallback(node.Edgenode.GetID(), nil, state.InfoCallback()); err != nil {
+		return "", fmt.Errorf("fail in get InfoLastCallback: %w", err)
+	}
+	if err := ctrl.MetricLastCallback(node.Edgenode.GetID(), nil, state.MetricCallback()); err != nil {
+		return "", fmt.Errorf("fail in get MetricLastCallback: %w", err)
+	}
+	appStatesSlice := make([]*eve.AppInstState, 0, len(state.Applications()))
+	appStatesSlice = append(appStatesSlice, state.Applications()...)
+	for _, app := range appStatesSlice {
+		if app.Name == appName {
+			return app.EVEState, nil
+		}
+	}
+
+	return "", fmt.Errorf("app %s not found", appName)
+}
+
+// AppSSHExec executes a command on the app VM running on the EVE node
+func (node *RemoteNode) AppSSHExec(command string) (string, error) {
```

Comment:
Which app? Why is this method not taking appName as an argument?

Reply:
I was operating on the fact that my test only needs one app, but right I will change it to accept app name.